### ISSUE TITLE
Error creating group

### ISSUE THREAD

**Issue by vishalvinayak** [HDFGroup/h5serv#110]
Hi,

Just getting into h5serv. Trying to figure out what's the best way to create a group. I'm using postman and trying with the following URI:

http://<my_website>/groups?host=<group_name>.<domain_name>.<user>.home.<my_website>
and I have basic auth set up.

I get a "tornado.web.HTTPError: HTTP 404: Not Found" error which means it's not able to find the file. I tried with the following too:
http://<my_website>/groups

and the following in the params:
host=<group_name>.<domain_name>.<user>.home.<my_website>

still get the same error.

Can anyone explain with an example what's the correct way to create a group? Thanks in advance.

**Comment by jreadey** [comment]
Hi @vishalvinayak,
  That doesn't look quite right - the group name isn't pre-appended to the domain name.

   See the docs here: http://h5serv.readthedocs.io/en/latest/GroupOps/POST_Group.html. 

   And there are some examples in h5serv/test/integ/grouptest.py.

    Also, if you want the domain names to end with <my.website>, you'll need to change the h5serv/server/config.py "domain" setting.

**Comment by nsingla** [comment]
Hi @jreadey ,

Where do you specify the group name then? In the documentation below is testGroupPostWithLink is just a domain name(file name)? In the documentation we've been getting confused about some examples referencing to tall.test where i assume test is the domain name, not sure in this one though.

We already configured the config.py to make domain work. That works perfectly.

POST /groups HTTP/1.1
Content-Length: 79
User-Agent: python-requests/2.3.0 CPython/2.7.8 Darwin/14.0.0
host: testGroupPostWithLink.test.hdfgroup.org
Accept: */*
Accept-Encoding: gzip, deflate

**Comment by jreadey** [comment]
By itself POST /groups just create an anonymous group (i.e. not connected with any other objects in the domain).  To link it at the same provide a body with the request which includes the name of the link and the id that it should be linked from.  See the "Sample Request with Link" example request and response.

You can spy on the data directory and run tools like "h5dump" to verify this all went according to plan. 

If you are not familiar with Python, it maybe a little hard to follow the logic in the test cases, but shouldn't be too bad.

**Comment by ghost** [comment]
@jreadey : i think the test scripts in the test folder are very helpful. We're going to reference those going forward, appreciate the help!!

**Comment by jreadey** [comment]
Out of curiosity - what language are you using?

**Comment by ghost** [comment]
all python, we're trying to use hdf5 instead of postgresql. We really like the h5serv concept for the kind of stuff we're doing.

**Comment by jreadey** [comment]
Great - let me know how it goes.

BTW - We've been working on a AWS S3-backed version of HDF Server - you can read about it here: https://s3.amazonaws.com/hdfgroup/docs/HDF+Scalable+Data+Service.pdf.  It's not open source yet though.

**Comment by jreadey** [comment]
Closing the issue.
